src/components/Todo/Todo.js

```diff
@@ -1,70 +1,170 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState, useEffect, useCallback } from 'react';
 import InputItem from '../InputItem/InputItem';
 import ItemList from '../ItemList/ItemList';
 import Footer from '../Footer/Footer';
 import styles from './Todo.module.css';
+import { ItemsContext } from '../../itemsContext';
+import { DndProvider } from 'react-dnd';
+import { HTML5Backend } from 'react-dnd-html5-backend';
+import update from 'immutability-helper';
+
+const initialItems = []
+
+const initialCount = initialItems.length;
+
+const saveToLocalStorage = (items) => {
+  let serialItems = JSON.stringify(items);
+  localStorage.setItem("items", serialItems);
+}
 
 function Todo() {
-  useEffect(() => {console.log('update')});
+  let savedItems, savedCount;
+  if (localStorage.length) {
+    savedItems = JSON.parse(localStorage.getItem("items"));
+    savedCount = parseInt(localStorage.getItem("count"), 10);
+  } else {
+    saveToLocalStorage(initialItems);
+    localStorage.setItem("count", initialCount);
+    savedItems = initialItems;
+    savedCount = initialCount;
+  }
+  useEffect(() => {});
   useEffect(() => {console.log('mount')},[]);
-  const [count, setCount] = useState(4);
-  const [items, setItems] = useState(
-    [
-      {
-        value: 'Сделать урок по React',
-        isDone: false,
-        id: 1
-      },
-      {
-        value: 'Приготовить обед',
-        isDone: false,
-        id: 2
-      },
-      {
-        value: 'Помыть посуду',
-        isDone: false,
-        id: 3
-      },
-      {
-        value: 'Погулять с собакой',
-        isDone: false,
-        id: 4
-      }
-    ]);
+  const [bottomNavigationValue, setBottomNavigationValue] = useState("all");
+  const [count, setCount] = useState(savedCount);
+  const [items, setItems] = useState(savedItems);
+
+  const moveItems = useCallback(
+    (dragIndex, hoverIndex) => {
+      const dragCard = items[dragIndex]
+      const newItemList = update(items, {
+        $splice: [
+          [dragIndex, 1],
+          [hoverIndex, 0, dragCard],
+        ],
+      });
+      setItems(newItemList);
+      saveToLocalStorage(newItemList);
+    },
+    [items],
+  );
 
   const onClickDone = id => {
     const newItemList = items.map(item => {
       const newItem = { ...item };
       if (item.id === id) {
         newItem.isDone = !item.isDone;
+        if (bottomNavigationValue != "all") newItem.hide = !item.hide;
```

Comment:
Codacy found an issue: [Expected '!==' and instead saw '!='.](https://app.codacy.com/gh/WebHeroSchool/agatta17_todo/pullRequest?prid=6081322)